src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryTable.cs

```diff
@@ -441,10 +442,14 @@ public static async ValueTask<FormulaValue> SortTable(EvalVisitor runner, EvalVi
             {
                 return SortValueType<DateTimeValue, DateTime>(pairs, irContext, compareToResultModifier);
             }
-            else
+            else if (allDates)
             {
                 return SortValueType<DateValue, DateTime>(pairs, irContext, compareToResultModifier);
             }
+            else
```

Comment:
else if (allOptionSets)?
And return an ErrorValue of some sort if no conditions match